components/gitpod-db/src/typeorm/migration/1675744891502-WorkspaceOrganizationId.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { log } from "@gitpod/gitpod-protocol/lib/util/logging";
+import { MigrationInterface, QueryRunner } from "typeorm";
+import { columnExists, indexExists } from "./helper/helper";
+
+const table = "d_b_workspace";
+const column = "organizationId";
+const orgIndex = "idx_organizationId";
+
+export class WorkspaceOrganizationId1675744891502 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        if (!(await columnExists(queryRunner, table, column))) {
+            await queryRunner.query(
+                `ALTER TABLE ${table} ADD COLUMN ${column} varchar(255), ALGORITHM=INPLACE, LOCK=NONE`,
```

Comment:
Thinking about this, I feel we should address the type change now, as it can impact performance severely. Plus d_b_workspace is our (2nd) biggest table, which we had to fixup in the past already for this reason. :+1:

Reply:
> tested a MAP_NULL_TO_UNDEFINED transformer in the follow-up PR, which works fine.

That would be awesome, looking forward to that!

Reply:
> Another thought I had after @selfcontained 's comment yesterday: It would be a great start to separate schema changes and data migrations into separate files (and thus: transactions), so we get more predictable behavior. But that's also sth for the future.

I believe the migration consists of the introduction of the column *and* setting its value. So running it one transaction and having it co-located seems reasonable. For me the reason to split it up would be runtime behavior that would cause bad side-effects when the migration runs, i.e. long running transactions.